platform/qt/src/qmapboxgl.cpp

```diff
@@ -785,6 +894,21 @@ void QMapboxGL::setLayoutProperty(const QString& layer_, const QString& property
     }
 }
 
+/*!
+    Sets a paint \a property \a value to an existing \a layer_. The \a property string can be any
+    as defined by \l {https://www.mapbox.com/mapbox-gl-style-spec/} {the vector tiles specification}
+    for paint properties.
+
+    The argument \a klass_ is deprecated and is used for defining the style class for the paint
+    property.
+
+    For paint properties that take a color as \a value, such as \b "fill-color", a string such as
+    \b "blue" can be passed or a QColor.
```

Comment:
Are there any other special cases? Perhaps a mapping from style specification types to Q-types would be appropriate here. This guide doesn’t have to be as longwinded as #7542, but something simple could head off newbie mistakes. I’m curious, for example, about the type `line-dasharray` takes.